documents/coredocument.go

```diff
@@ -208,17 +208,27 @@ func (cd *CoreDocument) PrepareNewVersion(documentPrefix []byte, collaborators C
 
 	err = populateVersions(&cdp, &cd.Document)
 	if err != nil {
-		return nil, err
+		return nil, errors.NewTypedError(ErrCDNewVersion, err)
 	}
 
 	ncd := &CoreDocument{Document: cdp}
 	ncd.addCollaboratorsToReadSignRules(rcs)
 	ncd.addCollaboratorsToTransitionRules(documentPrefix, wcs)
+	ncd.Attributes = copyAttrMap(cd.Attributes)
 	ncd.Modified = true
 	return ncd, nil
 
 }
 
+// copyAttrMap copies the attributes map
+func copyAttrMap(attributes map[string]*attribute) map[string]*attribute {
+	m := make(map[string]*attribute)
+	for k, v := range attributes {
+		m[k] = v
```

Comment:
This is still a shallow copy since attribute `v` is a pointer. Lets make a copy of the value attribute and assign its pointer